datahub-actions/src/datahub_actions/plugin/action/propagation/docs/propagation_action.py

```diff
@@ -0,0 +1,645 @@
+                logger.debug("Processing EntityChangeEvent Documentation Change")
+                if self.config.columns_enabled and (
+                    semantic_event.entityType == "schemaField"
+                ):
+                    if semantic_event.parameters:
+                        parameters = semantic_event.parameters
+                    else:
+                        parameters = semantic_event._inner_dict.get(
+                            "__parameters_json", {}
+                        )
+                    doc_string = parameters.get("description")
+                    origin = parameters.get("origin")
+                    origin = origin or semantic_event.entityUrn
+                    via = (
+                        semantic_event.entityUrn
+                        if origin != semantic_event.entityUrn
+                        else None
+                    )
+                    logger.debug(f"Origin: {origin}")
+                    logger.debug(f"Via: {via}")
+                    logger.debug(f"Doc string: {doc_string}")
+                    logger.debug(f"Semantic event {semantic_event}")
+                    if doc_string:
+                        return DocPropagationDirective(
+                            propagate=True,
+                            doc_string=doc_string,
+                            operation=semantic_event.operation,
+                            entity=semantic_event.entityUrn,
+                            origin=origin,
+                            via=via,  # if origin is set, then via is the entity itself
+                            actor=(
+                                semantic_event.auditStamp.actor
+                                if semantic_event.auditStamp
+                                else self.actor_urn
+                            ),
+                        )
+        return None
+
+    def modify_docs_on_columns(
+        self,
+        graph: AcrylDataHubGraph,
+        operation: str,
+        schema_field_urn: str,
+        dataset_urn: str,
+        field_doc: Optional[str],
+        context: SourceDetails,
+    ) -> Optional[MetadataChangeProposalWrapper]:
+        if context.origin == schema_field_urn:
+            # No need to propagate to self
+            return None
+
+        if not dataset_urn.startswith("urn:li:dataset"):
+            logger.error(
+                f"Invalid dataset urn {dataset_urn}. Must start with urn:li:dataset"
+            )
+            return None
+
+        auditStamp = AuditStampClass(
+            time=int(time.time() * 1000.0), actor=self.actor_urn
+        )
+
+        source_details = context.dict(exclude_none=True)
+        attribution: MetadataAttributionClass = MetadataAttributionClass(
+            source=self.action_urn,
+            time=auditStamp.time,
+            actor=self.actor_urn,
+            sourceDetail=source_details,
+        )
+        documentations = graph.graph.get_aspect(schema_field_urn, DocumentationClass)
```

Comment:
not a huge fan of this `graph.graph.method` :( probably not worth fixing right now